_source/_posts/2022-09-26-java-jetty-oauth.md

```diff
@@ -0,0 +1,1017 @@
+
+You need to run the following commands from the `spring-boot-jetty-maven-okta` subdirectory.
+
+{% include setup/cli.md type="web" framework="Okta Spring Boot Starter" loginRedirectUri="https://oidcdebugger.com/debug,http://localhost:8080/login/oauth2/code/okta" %}
+
+> **NOTE:** You will use the oidcdebugger.com redirect URI to create an access token you can use from the command line with HTTPie. The second URI is the default redirect URI that Spring Security uses for Okta when using its OAuth login feature.
+
+## Configure the Spring Boot app for OIDC authentication
+
+The project uses [the Okta Spring Boot Starter](https://github.com/okta/okta-spring-boot) to simplify configuring the web server. Find the following block in the `pom.xml` file and uncomment it.
+
+```xml
+<dependency>
+    <groupId>com.okta.spring</groupId>
+    <artifactId>okta-spring-boot-starter</artifactId>
+    <version>2.1.6</version>
+</dependency>
+```
+
+Confirm that your issuer URI, client ID, and client secret were added to the `src/main/resources/application.properties` file.
+
+```properties
+okta.oauth2.issuer=https://{yourOktaDomain}/oauth2/default
+okta.oauth2.clientId={clientId}
+okta.oauth2.clientSecret={clientSecret}
+```
+
+You also need to update the `SpringBootJettyApplication` class to match the following:
+
+```java
+package com.demo;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+@EnableWebSecurity
+public class SpringBootJettyApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(SpringBootJettyApplication.class, args);
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .anyRequest().permitAll()
+                .and()
+                .oauth2ResourceServer().jwt();
+        return http.build();
+    }
+
+}
+```
+
+The above class enables JWT-based OAuth and OIDC security. However, it explicitly allows any request. You are doing this because you are going to use method-level security.
+
+## Protect the DELETE and POST endpoints
+
+Add the `@PreAuthorize("isAuthenticated")` annotation to the `indexPost()` and `indexDelete()` methods of the `WebController` class. This will require that each request to those endpoints be authenticated -- that is, a valid user will need to be logged in. Because you are using Okta as your OIDC provider, they will be able to have authenticated with Okta's servers, typically through a single sign-on.
```

Comment:
```suggestion
Add the `@PreAuthorize("isAuthenticated")` annotation to the `indexPost()` and `indexDelete()` methods in the `WebController` class. This will require that each request to those endpoints be authenticated -- that is, a valid user will need to be logged in. Because you are using Okta as your OIDC provider, they will be able to have authenticated with Okta's servers, typically through a single sign-on.
```